python/ray/experimental/data/dataset.py

```diff
@@ -359,46 +363,124 @@ def write_parquet(path: str,
         Time complexity: O(dataset size / parallelism)
 
         Args:
-            path: The path in the filesystem to write to.
+            path: The path to the destination root directory, where Parquet
+                files will be written to..
             filesystem: The filesystem implementation to write to.
         """
-        raise NotImplementedError  # P0
+        import pyarrow.parquet as pq
+
+        @ray.remote
+        def parquet_write(write_path, block):
+            print(f"Writing {block.num_rows()} records to {write_path}.")
+            with pq.ParquetWriter(write_path, block._table.schema) as writer:
+                writer.write_table(block._table)
 
-    def write_json(path: str,
-                   filesystem: Optional["pyarrow.fs.FileSystem"] = None
-                   ) -> None:
+        refs = [
+            parquet_write.remote(
+                os.path.join(path, f"data{block_idx}.parquet"), block)
+            for block_idx, block in enumerate(self._blocks)
+        ]
+
+        # Block until writing is done.
+        ray.get(refs)
+
+    def write_json(
+            self,
+            paths: Union[str, List[str]],
+            filesystem: Optional["pyarrow.fs.FileSystem"] = None) -> None:
         """Write the dataset to json.
 
         This is only supported for datasets convertible to Arrow records.
 
         Examples:
+            # Write to a single file.
             >>> ds.write_json("s3://bucket/path")
 
+            # Write to multiple files.
+            >>> ds.write_json(["/path/to/file1", "/path/to/file2"])
```

Comment:
Remove this (same semantics as write_parquet).

Reply:
I'll make this change for now, but I think that we should try to find a better API in the future. Blocks are an implementation detail of `Dataset`, so I find it a bit weird that we're forcing the written files to be at the block granularity instead of providing e.g. a `num_output_partitions` API.

Reply:
We can revisit if we want to add partitioning support to the writer in the future.